runtime/tests/runtime_integration_tests/test_fe_programming.rs

```diff
@@ -79,3 +79,58 @@ fn test_fe_programming_invalid_partition() {
         ))
     );
 }
+
+#[test]
+fn test_fe_programming_cmd_warm_reset() {
+    let rom = caliptra_builder::rom_for_fw_integration_tests().unwrap();
+    let init_params = InitParams {
+        rom: &rom,
+        security_state: *SecurityState::default().set_device_lifecycle(DeviceLifecycle::Production),
+        ..Default::default()
+    };
+
+    let mut model = run_rt_test(RuntimeTestArgs {
+        init_params: Some(init_params),
+        ..Default::default()
+    });
+
+    model.step_until(|m| {
+        m.soc_ifc().cptra_boot_status().read() == u32::from(RtBootStatus::RtReadyForCommands)
+    });
+
+    // Create FE programming request with test parameters.
+    let mut cmd = MailboxReq::FeProg(FeProgReq {
+        hdr: MailboxReqHeader { chksum: 0 },
+        partition: 1,
+    });
+    cmd.populate_chksum().unwrap();
+
+    let resp = model
+        .mailbox_execute(u32::from(CommandId::FE_PROG), cmd.as_bytes().unwrap())
+        .unwrap()
+        .expect("We should have received a response");
+
+    // Verify we got a successful response (should be at least header size)
+    assert!(resp.len() >= core::mem::size_of::<MailboxReqHeader>());
```

Comment:
Should this be referring the response header? Same in the command below.

```Rust
    assert!(resp.len() >= core::mem::size_of::<MailboxRespHeader>());
```

Reply:
Yeah, I think you're right